SA3/qa_web/views.py

```diff
@@ -384,3 +383,28 @@ def pagination_data(self, paginator, page, is_paginated):
 
         return context
 
+
+# edit post
+@login_required(login_url='/login/')
+def edit(request, id_):
+    if id_:
```

Comment:
I am not sure this if-else block will accomplish the right thing. If the id_ is a false-y value such as None or 0, it will create a new Question but will still return a 404 response on line 397. This could lead to creating many empty questions in the database by simply refreshing the url `/questions/0/edit/`. We might just want to drop the if/else altogether, the `get_object_or_404` seems like sufficient protection against invalid `id_` values.